cmd/fluxcli/main.go

```diff
@@ -0,0 +1,316 @@
+	"github.com/pkg/errors"
+)
+
+var (
+	q     = flag.String("q", "", "flux script")
+	httpd = flag.String("http", "", "run http server on specified port")
+)
+
+func main() {
+	flag.Parse()
+
+	if *httpd != "" {
+		runHttp()
+		return
+	}
+
+	if *q == "" {
+		fmt.Println("query required")
+		os.Exit(1)
+	}
+
+	c := lang.FluxCompiler{
+		Query: *q,
+	}
+	d := csv.DefaultDialect()
+
+	querier := NewQuerier()
+
+	var buf bytes.Buffer
+	_, err := querier.Query(context.Background(), &buf, c, d)
+	if err != nil {
+		log.Fatalf("failed to run query: %v", err)
+		os.Exit(1)
+	}
+
+	fmt.Print(buf.String())
+}
+
+func runHttp() {
+	querier := NewQuerier()
+
+	http.HandleFunc("/v2/query", func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+
+		req, err := decodeQueryRequest(r)
+		if err != nil {
+			log.Println("error decoding request:", err)
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+
+		log.Printf("handling request: %s\n", req.Query)
+
+		pr := req.ProxyRequest()
+		q, err := querier.c.Query(ctx, pr.Request.Compiler)
+		if err != nil {
+			log.Println("error executing query:", err)
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+		defer func() {
+			q.Cancel()
+			q.Done()
+		}()
+
+		// Setup headers
+		//stats, hasStats := results.(flux.Statisticser)
+		//if hasStats {
+		//	w.Header().Set("Trailer", statsTrailer)
+		//}
+
+		// NOTE: We do not write out the headers here.
+		// It is possible that if the encoding step fails
+		// that we can write an error header so long as
+		// the encoder did not write anything.
+		// As such we rely on the http.ResponseWriter behavior
+		// to write an StatusOK header with the first write.
+
+		switch r.Header.Get("Accept") {
+		case "text/csv":
+			fallthrough
+		default:
+
+			if hd, ok := pr.Dialect.(httpDialect); !ok {
+				log.Printf("unsupported dialect %T\n", req.Dialect)
+				http.Error(w, fmt.Sprintf("unsupported dialect over HTTP %T", req.Dialect), http.StatusBadRequest)
+				return
+			} else {
+				hd.SetHeaders(w)
+			}
+			encoder := pr.Dialect.Encoder()
+			results := flux.NewResultIteratorFromQuery(q)
+			n, err := encoder.Encode(w, results)
+			if err != nil {
+				results.Cancel()
+				if n == 0 {
+					log.Println("no results:", err)
+					// If the encoder did not write anything, we can write an error header.
+					http.Error(w, err.Error(), http.StatusInternalServerError)
+				}
+			}
+		}
+	})
+
+	log.Printf("http server listening %s\n", *httpd)
+
+	http.ListenAndServe(*httpd, nil)
+}
+
+type Querier struct {
+	c *control.Controller
+}
+
+func (q *Querier) Query(ctx context.Context, w io.Writer, c flux.Compiler, d flux.Dialect) (int64, error) {
+	qry, err := q.c.Query(ctx, c)
+	if err != nil {
+		return 0, err
+	}
+	results := flux.NewResultIteratorFromQuery(qry)
+	defer results.Cancel()
+
+	encoder := d.Encoder()
+	return encoder.Encode(w, results)
+}
+
+func NewQuerier() *Querier {
```

Comment:
How does this daemon have any data to return? Is the expectation that you use sources that reach out to other systems? How are those sources configured?

Reply:
yes, that's correct.  it assumes that the source you use will be able to read from its intended source without any support from this binary.  Not sure what would happen is a user tried to use influxdb.from

Reply:
the built-in sources should be self contained, such that calling `fromCSV` or `fromSQL` in a Flux query should just execute.

For example, the path to `fromCSV` will serve up files from the local file system

Reply:
@aanthony1243 I think `from` to InfluxDB will be removed from this repository, however, there isn't anything stopping us adding `fromInfluxDB` in the future that talks over gRPC